Problem3.rb

```diff
@@ -0,0 +1,48 @@
+require 'rspec'
+
+def swap(array, i, j)
+
+  temp = array[i]
+  array[i] = array[j]
+  array[j] = temp
+
+end
+
+def contains_duplicates?(array)
+
+  min = 1
+
+  array.each_index do |i|
+
+    temp = array[i]
+    j = temp - min
+
+    while i != j do
+
+      if temp == array[j]
+        return true
+      end
+
+     swap(array, i, j)
+
+      j = temp - min
+
+    end

```

Comment:
We write code for people, not machines. Spent ~ 15 minutes to figure out what the code does, but still a mistery to me. Can you please describe what you wanted to do ?

Btw, it is not 100% working

![image](https://user-images.githubusercontent.com/2707601/39755024-efb859b8-52cc-11e8-8fbe-8669bd3942c9.png)

Reply:
I completly re-implemented problem3 to meet the O(n) requirements

Reply:
Algorithm fails because of the swap method.